pixy/testutils.go

```diff
@@ -97,3 +111,87 @@ func (tkc *TestKafkaClient) getMessages(topic string, before, after []int64) [][
 	}
 	return writtenMsgs
 }
+
+// GenMessage generates an ASCII message of the specified size.
+func GenMessage(size int) string {
+	b := bytes.NewBuffer(nil)
+	for b.Len() < size {
+		b.WriteString(strconv.Itoa(b.Len()))
+		b.WriteString("-")
+	}
+	return string(b.Bytes()[:size])
+}
+
+// ChunkReader allows reading its underlying buffer in chunks making the
+// specified pauses between the chunks. After each pause `Read()` returns
+// `0, nil`. This kind of reader is useful to simulate HTTP requests that
+// require several read operations on the request body to get all of it.
+type ChunkReader struct {
+	chunks   []string
+	chunk    string
+	pause    time.Duration
+	chunkDue time.Time
+}
+
+func NewChunkReader(s string, count int, pause time.Duration) *ChunkReader {
```

Comment:
I fixed the caller to have non zero value. Yet another case of me forgetting to revert a temporary change.